R/mod_view_tables.R

```diff
@@ -0,0 +1,270 @@
+          table_info$edit_info$value,
+          " WHERE rowid = ",
+          table_info$data$row_id[table_info$data$row_number == table_info$edit_info$row]
+        )
+    }
+    else{
+      update_query <-
+        paste0(
+          "UPDATE ",
+          conn$active_table,
+          " SET ",
+          table_info$column_names$name[table_info$edit_info$col + 1],
+          " = '",
+          table_info$edit_info$value,
+          "' WHERE rowid = ",
+          table_info$data$row_id[table_info$data$row_number == table_info$edit_info$row]
+        )
+    }
+    # print(update_query)
+    # print(class(table_info$edit_info$value))
+    tryCatch(
+      expr = {
+        RSQLite::dbExecute(conn$active_db, update_query)
+        table_info$data[table_info$edit_info$row, table_info$edit_info$col +
+                          3] <- table_info$edit_info$value
+      },
+      error = function(err) {
+        showNotification(
+          ui =  paste0(err, ". Changes not saved."),
+          duration = 15,
+          type = "error"
+        )
+        data_fetch_query <-
+          paste0(
+            "SELECT rowid AS row_id, ROW_NUMBER() OVER(ORDER BY rowid) AS row_number, * FROM ",
+            conn$active_table
+          )
+        table_info$data <-
+          RSQLite::dbGetQuery(conn$active_db, data_fetch_query)
+      }
+    )
+  })
+  
+  observeEvent(input$delete_rows, {
+    if (is.null(input$display_table_rows_selected)) {
+      showNotification(ui =  "No rows selected.",
+                       duration = 3,
+                       type = "error")
+    }
+    else{
+      table_info$page <- input$display_table_rows_current[1] - 1
+      info <- input$display_table_rows_selected
+      # print(input$display_table_rows_selected[1])
+      for (i in info) {
+        delete_query <-
+          paste0("DELETE FROM ",
+                 conn$active_table,
+                 " WHERE rowid = ",
+                 table_info$data$row_id[table_info$data$row_number == i])
+        RSQLite::dbExecute(conn$active_db, delete_query)
+      }
+      data_fetch_query <-
+        paste0(
+          "SELECT rowid AS row_id, ROW_NUMBER() OVER(ORDER BY rowid) AS row_number, * FROM ",
+          conn$active_table
+        )
+      table_info$data <-
+        RSQLite::dbGetQuery(conn$active_db, data_fetch_query)
+      showNotification(ui =  "Selected rows deleted successfully.",
+                       duration = 3,
+                       type = "message")
+    }
+  })
+  
+  observeEvent(input$display_table_rows_selected,{
+    print(input$display_table_rows_selected)
+    # table_info$page <- input$display_table_rows_current[1] - 1
+  })
+  
+  observeEvent(input$delete_all_rows, {
+    delete_query <-
+      paste0("DELETE FROM ",
+             conn$active_table)
+    RSQLite::dbExecute(conn$active_db, delete_query)
+    
+    data_fetch_query <-
+      paste0(
+        "SELECT rowid AS row_id, ROW_NUMBER() OVER(ORDER BY rowid) AS row_number, * FROM ",
```

Comment:
How about storing all the query strings in another R script and sourcing it here?